File: program_data/github_java_sort_function/6/30.java
package array._20170807;

import java.util.Arrays;


public class RadixSort {

    
    public static void sort(int[] arr) {
        int m = getMax(arr);
        for (int exp = 1; m / exp > 0; exp *= 10) {
            countSort(arr, exp);
        }
        System.out.println(Arrays.toString(arr));
    }

    private static void countSort(int[] arr, int exp) {
        int length = arr.length;
        int[] output = new int[length];
        int[] count = new int[10];
        for (int i = 0; i < length; i++) {
            count[(arr[i] / exp) % 10]++;
        }
        for (int i = 1; i < 10; i++) {
            count[i] += count[i - 1];
        }
        for (int i = length - 1; i >= 0; i--) {
            output[count[(arr[i] / exp) % 10] - 1] = arr[i];
            count[(arr[i] / exp) % 10]--;
        }
        for (int i = 0; i < length; i++) {
            arr[i] = output[i];
        }
    }

    private static int getMax(int[] arr) {
        int max = arr[0];
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] > max) {
                max = arr[i];
            }
        }
        return max;
    }


}

File: program_data/PKU_raw/12/30.c
int main()
{
	int a[20];
	int i=0,n=0,j;
	int count=0;
	do
	{
		n=0;
		count=0;
		scanf("%d",&a[0]);
		if(a[0]!=-1)
		{
			i=1;
			do
			{
				scanf("%d",&a[i]);
				i++;
				count++;
			}while(a[i-1]!=0);

			for(i=0;i<count;i++)
			{
				for(j=i+1;j<count;j++)
				{
					if(a[i]==2*a[j] || 2*a[i]==a[j])
					{
						n++;
					}
				}
			}
			printf("%d\n",n);
		}

	}while(a[0]!=-1);
	return 0;
}